cql3/statements/select_statement.cc

```diff
@@ -956,14 +958,31 @@ lw_shared_ptr<const service::pager::paging_state> indexed_table_select_statement
         throw exceptions::invalid_request_exception("Indexed column not found in schema");
     }
 
-    auto result_view = query::result_view(*results);
-    if (!results->row_count() || *results->row_count() == 0) {
-        return paging_state;
+    if (!results->is_short_read() && (!results->row_count() || *results->row_count() == 0)) {
+        auto paging_state_copy = make_lw_shared<service::pager::paging_state>(service::pager::paging_state(*paging_state));
+        paging_state_copy->set_remaining(0);
```

Comment:
Why wasn't this set_remaining(0) needed in the old code, and why is it needed now?

Reply:
hmm, I think I meant it as a safety net at some point, but it's probably not needed. It could be an assert.